runners/java-job-service/src/main/java/org/apache/beam/runners/jobsubmission/InMemoryJobService.java

```diff
@@ -287,26 +283,50 @@ public void run(RunJobRequest request, StreamObserver<RunJobResponse> responseOb
     }
   }
 
+  private Map<String, List<RunnerApi.ArtifactInformation>> extractDependencies(
```

Comment:
Can this return/be an ImmutableMap

Reply:
Same as above. (I suppose here the method is private, but it's stell IMHO preferable to not declare implementation details in the signature.)